src/stdlib_hashmap_open.f90

```diff
@@ -0,0 +1,875 @@
+                map % inverse( inmap ) % target => new_ent
+                map % slots( test_slot ) = inmap
+                if ( present(conflict) ) conflict = .false.
+                return
+            else if ( inmap < 0 .or. &
+                inmap > map % num_entries + map % num_free ) then
+                error stop submodule_name // ' % ' // procedure // ': ' // &
+                    invalid_inmap
+            else if (.not. associated( map % inverse(inmap) % target ) ) then
+                error stop submodule_name // ' % ' // procedure // ': ' // &
+                    invalid_inmap
+            else
+                associate( target => map % inverse(inmap) % target )
+                  if ( hash_val == target % hash_val ) then
+                      if ( key == target % key ) then
+                          ! entry already exists
+                          if ( present(conflict) ) then
+                              conflict = .true.
+                          else
+                              error stop submodule_name // ' % ' // procedure &
+                                  // ': ' // conflicting_key
+                          end if
+                          return
+                      end if
+                  end if
+                end associate
+            end if
+            offset = offset + 1
+        end do PROBE_SUCCESSIVE_SLOTS
+
+    contains
+
+        subroutine allocate_open_map_entry(map, bucket)
+!         allocates a hash bucket
+            type(open_hashmap_type), intent(inout) :: map
+
+            type(open_map_entry_type), pointer, intent(out) :: bucket
+            type(open_map_entry_list), pointer :: free_list
+            type(open_map_entry_pool), pointer :: pool
+
+            pool => map % cache
+            map % num_entries = map % num_entries + 1
+            if ( associated(map % free_list) ) then
+!             Get hash bucket from free_list
+                free_list => map % free_list
+                bucket => free_list % target
+                map % free_list => free_list % next
+                free_list % target => null()
+                free_list % next => null()
+                if (bucket % inmap == 0) stop "bucket % inmap == 0"
+                map % num_free = map % num_free - 1
+            else
+!             Get hash bucket from pool
+                if ( pool % next == pool_size ) then
+!         Expand pool
+                    call extend_map_entry_pool(map % cache)
+                    pool => map % cache
+                end if
+                bucket      => pool % more_map_entries(pool % next)
+                pool % next =  pool % next + 1 ! 0s based -> post-increment
+                if ( map % num_entries >                     &
+                     size( map % inverse, kind=int_index ) ) then
+                    call expand_inverse( map )
+                end if
+                if ( map % num_entries == 0 ) stop "MAP % NUM_ENTRIES == 0."
```

Comment:
[Same as #r865894471](https://github.com/fortran-lang/stdlib/pull/611/files#r865894471)

Reply:
This is in a section of code where we are attempting to expand the entry pool because it is too small for the number of entries. It cannot be too small if `map % num_entries == 0`, so either the logic of the rest of the code is wrong or there has been a bit error. Mind you we could detect more errors in the code if the comparison were `map % num_entries <= 0`.

Reply:
I have changed this in my copy and will check it in shortly.